tests/integration/targets/ec2_vpc_endpoint/tasks/main.yml

```diff
@@ -802,62 +805,30 @@
   # ============================================================
   # BEGIN POST-TEST CLEANUP
   always:
-  # Delete the routes first - you can't delete an endpoint with a route
-  # attached.
-  - name: Delete minimal route table (no routes)
-    ec2_vpc_route_table:
-      state: absent
-      lookup: id
-      route_table_id: '{{ rtb_creation_empty.route_table.id }}'
-    ignore_errors: true
-
-  - name: Delete minimal route table (IGW route)
-    ec2_vpc_route_table:
-      state: absent
-      lookup: id
-      route_table_id: '{{ rtb_creation_igw.route_table.id }}'
-    ignore_errors: true
-
-  - name: Delete endpoint
-    ec2_vpc_endpoint:
-      state: absent
-      vpc_endpoint_id: '{{ create_endpoint.result.vpc_endpoint_id }}'
-    ignore_errors: true
-
-  - name: Delete endpoint
-    ec2_vpc_endpoint:
-      state: absent
-      vpc_endpoint_id: '{{ create_rtb_endpoint.result.vpc_endpoint_id }}'
-    ignore_errors: true
-
-  - name: Query any remain endpoints we created (idempotency work is ongoing)  # FIXME
+  - name: Query any remain endpoints we created
     ec2_vpc_endpoint_info:
       query: endpoints
       filters:
         vpc-id:
         - '{{ vpc_id }}'
-    register: test_endpoints
+    register: remaining_endpoints
 
   - name: Delete all endpoints
     ec2_vpc_endpoint:
       state: absent
       vpc_endpoint_id: '{{ item.vpc_endpoint_id }}'
-    with_items: '{{ test_endpoints.vpc_endpoints }}'
-    ignore_errors: true
-
-  - name: Remove IGW
-    ec2_vpc_igw:
-      state: absent
-      vpc_id: '{{ vpc_id }}'
-    register: igw_deletion
-    retries: 10
-    delay: 5
-    until: igw_deletion is success
-    ignore_errors: yes
-
-  - name: Remove VPC
-    ec2_vpc_net:
-      state: absent
-      name: '{{ vpc_name }}'
-      cidr_block: '{{ vpc_cidr }}'
+      wait: true
+    loop: '{{ remaining_endpoints.vpc_endpoints }}'
     ignore_errors: true
+    register: endpoints_removed
+    until:
+      - endpoints_removed is not failed
+      - endpoints_removed is not changed
+    retries: 20
+    delay: 10
```

Comment:
"wait: true" is documented as only working for creation, not deletion :(  This isn't a bug, but would be a nice feature.  Out of scope for just getting the tests running.